habitat/datasets/vln/r2r_vln_dataset.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import gzip
+import json
+import os
+from typing import List, Optional
+
+from habitat.config import Config
+from habitat.core.dataset import Dataset
+from habitat.core.logging import logger
+from habitat.core.registry import registry
+from habitat.datasets.utils import VocabDict
+from habitat.tasks.nav.nav import NavigationGoal, ShortestPathPoint
+from habitat.tasks.vln.vln import InstructionData, VLNEpisode
+
+DEFAULT_SCENE_PATH_PREFIX = "data/scene_datasets/mp3d/"
+
+R2R_VAL_SEEN_EPISODES = 781
+
+
+@registry.register_dataset(name="R2RVLN-v1")
+class VLNDatasetV1(Dataset):
+    r"""Class inherited from Dataset that loads the MatterPort3D
+    Room-to-Room (R2R) dataset for Vision and Language Navigation.
+    """
+
+    episodes: List[VLNEpisode]
+    instruction_vocab: VocabDict
+
+    @staticmethod
+    def check_config_paths_exist(config: Config) -> bool:
+        return os.path.exists(
+            config.DATA_PATH.format(split=config.SPLIT)
+        ) and os.path.exists(config.SCENES_DIR)
+
+    def __init__(self, config: Optional[Config] = None) -> None:
+        self.episodes = []
+
+        if config is None:
+            return
+
+        dataset_filename = config.DATA_PATH.format(split=config.SPLIT)
+        with gzip.open(dataset_filename, "rt") as f:
+            self.from_json(f.read(), scenes_dir=config.SCENES_DIR)
+
+    def from_json(
+        self, json_str: str, scenes_dir: Optional[str] = None
+    ) -> None:
+
+        deserialized = json.loads(json_str)
+
+        # Done for the serialization test
```

Comment:
What is the reason that we need this if? Why we don't have `deserialized["instruction_vocab"]["word_list"]` in released version of dataset?